src/app/voice-assistant/voice-assistant-chat/chat-window/chat-window.component.ts

```diff
@@ -53,8 +56,25 @@ export class ChatWindowComponent implements OnInit {
 
             this.chatMessagesSubscription = this.chatService
                 .getChatMessagesObservable(chatId)
-                .pipe(map((messages) => messages.slice().reverse()))
-                .subscribe((messages) => (this.messages = messages));
+                .pipe(map((messages) => messages))
+                .subscribe(
+                    (messages) =>
+                        (this.messages = messages
+                            .filter((message, index, messages) => {
+                                if (index + 1 >= messages.length) {
+                                    return true;
+                                }
+                                if (
+                                    message.messageId ==
+                                    messages[index + 1].messageId
+                                ) {
+                                    return false;
+                                }
+                                return true;
+                            })
+                            .slice()
+                            .reverse()),
+                );
```

Comment:
Could this be done in the chat Service directly? Something like giving each `chatMessage` a `isStreaming` parameter and as long as the parameter is true never add the "new" message to the existing list of messages, but instead overwrite the last message in the list with its updated version.

Reply:
I have outsourced this filtering to a method located in chat.service